### PR TITLE
Add Azure AD oauth2. Close #585

### LINKED ISSUE #585: Authentication Office365
Hello, I wanted to know if you plan to implement authentication Office365, thank you very much, your project is very interesting.

### PR DESCRIPTION
**Pull request by gmist**

### CODE REVIEW

**Comment by lipis** [comment]
screen shot 2016-10-31 at 20 01 19

Am I doing something wrong?

File: main/auth/azure_ad.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+azure_ad_config = dict(
+  access_token_method='POST',
+  access_token_url='https://login.microsoftonline.com/common/oauth2/token',
+  authorize_url='https://login.microsoftonline.com/common/oauth2/authorize',
+  consumer_key=config.CONFIG_DB.azure_ad_client_id,
+  consumer_secret=config.CONFIG_DB.azure_ad_client_secret,
+  request_token_params={
+    'scope': 'openid, profile, user_impersonation'
```
Comment (lipis) [inline main/auth/azure_ad.py]: trailing comma

**Comment by gmist** [comment]
Hmm... strange... looks like you don't have administrator permissions for Azure AD

**Comment by lipis** [comment]
Can you also add it here: https://my-gae-init.appspot.com/signin/?

(I didn't find much time yet to enable my Azure AD to play around.. something is wrong there)

**Comment by lipis** [comment]
So shall I merge that?! Also was thinking maybe instead of `azure_ad` just `azure`? (for the variable at least.. or shall we keep it as is?)

**Comment by gmist** [comment]
```Azure Active Directory``` -> ```azure_ad``` I think it is best left as it is.

**Comment by lipis** [comment]
oki then :)

**Pull request merged**